crates/nargo/tests/test_data/simple_shield/Prover.toml

```diff
@@ -5,11 +5,7 @@ index = "0"
 note_hash_path = [
     "0x0000000000000000000000000000000000000000000000000000000000000000",
     "0x0e4223f3925f98934393c74975142bd73079ab0621f4ee133cee050a3c194f1a",
-    "0x2fd7bb412155bf8693a3bd2a3e7581a679c95c68a052f835dddca85fa1569a40"
+    "0x2fd7bb412155bf8693a3bd2a3e7581a679c95c68a052f835dddca85fa1569a40",
 ]
 to_pubkey_x = "0x0000000000000000000000000000000000000000000000000000000000000001"
 to_pubkey_y = "0x0000000000000002cf135e7506a45d632d270d45f1181294833fc48d823f272c"
-return = [
```

Comment:
you should not change the behaviour of the return

Reply:
same as above -- return is no longer needed since we are no longer applying constraints

Reply:
Using the return enforces the witness to be equal to this value, without adding a constraint.

Reply:
And if we are completely getting rid of the return in favor of constraining on some input. This circuit needs to be changed to pass in those return values (`nullifier_hash` and `receiver_note_commitment` I believe) or else the circuit will be unsafe

Reply:
The circuit can still be safe because if the verifier does not input the correct values (nullifier_hash and receiver_note_commitment), then it will fail verification. 

What we were doing before was more akin to testing IIUC. I will add it back as input in the form of `expected_nullifier_hash` and `expected_ receiver_note_commitment` to make it clear

Reply:
> What we were doing before was more akin to testing IIUC. I will add it back as input in the form of `expected_nullifier_hash` and `expected_ receiver_note_commitment` to make it clear

This is what I meant. If it is not added back as input does it not become unsafe?

Reply:
Its still safe in this case, as in the proof will not verify if you do not put the right value,its just that it now does not require any constraints to set the output of the program to be public.